library/std/src/sync/mpmc/list.rs

```diff
@@ -547,7 +547,7 @@ impl<T> Channel<T> {
         }
 
         let mut head = self.head.index.load(Ordering::Acquire);
-        let mut block = self.head.block.load(Ordering::Acquire);
+        let mut block = self.head.block.swap(ptr::null_mut(), Ordering::AcqRel);
```

Comment:
Is this worth a comment explaining why the `swap` part is important?

Reply:
The problem with the separate load/store was not _that_ subtle, but a comment wouldn't hurt.

Reply:
Subtle enough that whoever wrote this code and the reviewer missed it. ;)

Reply:
Done https://github.com/rust-lang/rust/pull/121778.